extensions/AzureCosmosDBMongoDB/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemory.cs

```diff
@@ -0,0 +1,309 @@
+        settings.ApplicationName = config.ApplicationName;
+        this._cosmosDBMongoClient = new MongoClient(settings);
+        this._cosmosMongoDatabase = this._cosmosDBMongoClient.GetDatabase(databaseName);
+
+        this._cosmosMongoDatabase.CreateCollectionAsync(collectionName);
+        this._cosmosMongoCollection = this._cosmosMongoDatabase.GetCollection<AzureCosmosDBMongoDBMemoryRecord>(collectionName);
+        this._collectionName = collectionName;
+    }
+
+    /// <inheritdoc />
+    public async Task CreateIndexAsync(string index, int vectorSize, CancellationToken cancellationToken = default)
+    {
+        var indexes = await this._cosmosMongoCollection.Indexes.ListAsync(cancellationToken).ConfigureAwait(false);
+        if (!indexes.ToList(cancellationToken).Any(index => index["name"] == index))
+        {
+            var command = new BsonDocument();
+            switch (this._config.Kind)
+            {
+                case AzureCosmosDBVectorSearchType.VectorIVF:
+                    command = this.GetIndexDefinitionVectorIVF(index, vectorSize);
+                    break;
+                case AzureCosmosDBVectorSearchType.VectorHNSW:
+                    command = this.GetIndexDefinitionVectorHNSW(index, vectorSize);
+                    break;
+            }
+
+            await this._cosmosMongoDatabase.RunCommandAsync<BsonDocument>(command, cancellationToken: cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<string>> GetIndexesAsync(CancellationToken cancellationToken = default)
+    {
+        var indexesCursor = await this._cosmosMongoCollection.Indexes.ListAsync(cancellationToken).ConfigureAwait(false);
+        var indexes = await indexesCursor.ToListAsync(cancellationToken).ConfigureAwait(false);
+
+        return indexes.Select(index => index["name"].AsString);
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteIndexAsync(string index, CancellationToken cancellationToken = default)
+    {
+        await this._cosmosMongoCollection.Indexes.DropOneAsync(index, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc />
+    public async Task<string> UpsertAsync(string index, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        AzureCosmosDBMongoDBMemoryRecord localRecord = AzureCosmosDBMongoDBMemoryRecord.FromMemoryRecord(record);
+        var filter = Builders<AzureCosmosDBMongoDBMemoryRecord>.Filter.Eq(r => r.Id, localRecord.Id);
+
+        var replaceOptions = new ReplaceOptions() { IsUpsert = true };
+
+        await this._cosmosMongoCollection.ReplaceOneAsync(filter, localRecord, replaceOptions, cancellationToken).ConfigureAwait(false);
+
+        return record.Id;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetSimilarListAsync(
+        string index,
+        string text,
+        ICollection<MemoryFilter>? filters = null,
```

Comment:
`filters` is not implemented. Does it mean the connector is still underdevelopment?

Reply:
The filters are in public preview now, I'll add it in the next iteration in a couple weeks. Does this block the PR right now?